src/components/FeatureViews/Notes/Notes.js

```diff
@@ -15,503 +17,462 @@ import EditIcon from "@material-ui/icons/EditOutlined";
-				break;
-			}
-			case "course": {
-				const note = {
-					body: noteBody,
-					complete: false,
-					course: ownerID,
-					important: notification,
-					title: noteTitle,
-				};
-				if (editID) {
-					api.patchCourseNote(editID, note, ownerType, ownerID);
-					setIsPost(false);
-				} else {
-					api.postCourseNote(note, ownerType);
-					setIsPost(true);
-				}
-				setSubmitting(true);
-				break;
-			}
-			default: {
-				const note = {
-					body: noteBody,
-					complete: false,
-					important: notification,
-					title: noteTitle,
-					user: ownerID,
-				};
-				if (editID) {
-					api.patchAccountNote(editID, note, ownerType, ownerID);
-					setIsPost(false);
-				} else {
-					api.postAccountNote(note, ownerType);
-					setIsPost(true);
-				}
-				setSubmitting(true);
-			}
-		}
-	}, [api, noteBody, notification, noteTitle, ownerID, ownerType, editID]);
+    // eslint-disable-next-line max-statements
+    const saveNote = useCallback(() => {
+        switch (ownerType) {
+            case "enrollment": {
+                const note = {
+                    "body": noteBody,
+                    "complete": false,
+                    "enrollment": ownerID.enrollmentID,
+                    "important": notification,
+                    "title": noteTitle,
+                };
+                if (editID) {
+                    api.patchEnrollmentNote(
+                        editID,
+                        note,
+                        ownerID.enrollmentID,
+                        ownerID.studentID,
+                        ownerID.courseID,
+                    );
+                    setIsPost(false);
+                } else {
+                    api.postEnrollmentNote(
+                        note,
+                        ownerID.enrollmentID,
+                        ownerID.studentID,
+                        ownerID.courseID,
+                    );
+                    setIsPost(true);
+                }
+                setSubmitting(true);
+                break;
+            }
+            case "course": {
+                const note = {
+                    "body": noteBody,
+                    "complete": false,
+                    "course": ownerID,
+                    "important": notification,
+                    "title": noteTitle,
+                };
+                if (editID) {
+                    api.patchCourseNote(editID, note, ownerType, ownerID);
+                    setIsPost(false);
+                } else {
+                    api.postCourseNote(note, ownerType);
+                    setIsPost(true);
+                }
+                setSubmitting(true);
+                break;
+            }
+            default: {
+                const note = {
+                    "body": noteBody,
+                    "complete": false,
+                    "important": notification,
+                    "title": noteTitle,
+                    "user": ownerID,
+                };
```

Comment:
Shouldn't we create a default note object and then add an additional key+pair to the note object depending on which case it is? (i.e. add "user" if it's an account note, add "course" if it's a course note, etc.)

Reply:
I'll change this in a later PR when implementing mutations.